hydra-node/src/Hydra/Logging.hs

```diff
@@ -128,19 +128,21 @@ withTracerOutputTo hdl namespace action = do
 -- metadata.
 showLogsOnFailure ::
   (MonadSTM m, MonadCatch m, MonadFork m, MonadTime m, MonadSay m, ToJSON msg) =>
+  Text ->
   (Tracer m msg -> m a) ->
   m a
-showLogsOnFailure action = do
+showLogsOnFailure namespace action = do
   tvar <- newTVarIO []
-  action (traceInTVar tvar)
+  action (traceInTVar tvar namespace)
     `onException` (readTVarIO tvar >>= mapM_ (say . decodeUtf8 . Aeson.encode) . reverse)
 
 traceInTVar ::
   (MonadFork m, MonadTime m, MonadSTM m) =>
   TVar m [Envelope msg] ->
+  Text ->
   Tracer m msg
-traceInTVar tvar = Tracer $ \msg -> do
-  envelope <- mkEnvelope "" msg
+traceInTVar tvar namespace = Tracer $ \msg -> do
+  envelope <- mkEnvelope namespace msg
```

Comment:
What is the benefit of having a namespace; where did we need it? I see it required a lot of code changes and makes the test boilerplate bigger.

Reply:
while working on the e2e test, we encountered a situation where resuming a Hydra node (after reconfiguring a party) became confusing. This confusion arose because the logs were overwritten by the second launch of the instance.
wdyt?

Reply:
It's not strictly needed, but having proper namespace to qualify the "entity" that logs is helpful, even if those logs are merely from the tester itself, it removes confusion.

Reply:
imo this tidy result from a need originated by this PR.

I can do it if you think differently, as it is self contained in a single commit so no big deal.

let me know your thoughts.